docs/operators/jdbcPollInputOperator.md

```diff
@@ -0,0 +1,175 @@
+=============
+
+## Operator Objective
+This operator scans JDBC database table in parallel fashion. This operator is added to address common input operator problems like,
+
+1. As discussed in [Development Best Practices](https://github.com/apache/apex-core/blob/master/docs/development_best_practices.md),
+    the operator callbacks such as `beginWindow()`, `beginWindow()`, `emitTuples()`, etc.
+    (which are invoked by the main operator thread)
+    are required to return quickly, well within the default streaming window duration of
+    500ms. This requirement can be an issue when retrieving data from slow external systems
+    such as databases or object stores: if the call takes too long, the platform will deem
+    the operator blocked and restart it. Restarting will often run into the same issue
+    causing an unbroken sequence of restarts.
+
+2. When a large volume of data is available from a single store that allows reading from
+   arbitrary locations (such as a file or a database table), reading the data sequentially
+   can be throughput limiting: Having multiple readers read from non-overlapping sections
+   of the store allows any downstream parallelism in the DAG to be exploited better to
+   enhance throughput. For files, this approach is used by the file splitter and block
+   reader operators in the Malhar library.
+
+JDBC Poller Input operator addresses the first issue with an asynchronous worker thread which retrieves the data and adds it to an in-memory queue; the main operator thread dequeue tuples very quickly if data is available or simply returns if not. The second is addressed in a way that parallels the approach to files by having multiple partitions read records from non-overlapping areas of the table. Additional details of how this is done are described below.
+
+#### Assumption
+Assumption is that there is an ordered column using which range queries can be formed. That means database has a column or combination of columns which has unique constraint as well as every newly inserted record should have column value more than max value in that column, as we poll only appended records.
+
+## Use cases
+1. Scan huge database tables to either copy to other database or process it using **Apache Apex**. An example application using this operator to copy database contents to HDFS is available in the [examples repository](https://github.com/DataTorrent/examples/tree/master/tutorials/jdbcIngest). Look for "PollJdbcToHDFSApp" for example of this particular operator.
+
+## How to Use?
+The tuple type in the abstract class is a generic parameter. Concrete subclasses need to choose an appropriate class (such as String or an appropriate concrete java class, having no argument constructor so that it can be serialized using kyro serializer, etc for the generic parameter). Also implement a couple of abstract methods: getTuple(ResultSet) to convert database rows to objects of concrete class and emitTuple(T) to emit the tuple.
```

Comment:
no argument => no-argument
serialized ... generic parameter =>  serialized using Kryo
Also, highlight literal function calls like getTuple(ResultSet) and emitTuple(T) using backquotes.